lib/ocpp/v201/charge_point.cpp

```diff
@@ -3161,6 +3162,19 @@ void ChargePoint::handle_set_charging_profile_req(Call<SetChargingProfileRequest
     SetChargingProfileResponse response;
     response.status = ChargingProfileStatusEnum::Rejected;
 
+    if (msg.chargingProfile.chargingProfilePurpose == ChargingProfilePurposeEnum::ChargingStationExternalConstraints) {
+        response.statusInfo = StatusInfo();
+        response.statusInfo->reasonCode = "InvalidValue";
```

Comment:
NIT: Could the reasonCode be InvalidProfileType? Not sure how we keep this information consistent.

Reply:
That isn't one of the accepted reason codes listed in the OCPP appendices